[MRG] Temporary disable MKL in Travis

lesteve opened the pull request:
to have master green on Travis until we fix #6279.

The main other change is to install nomkl when INSTALL_MKL is not true.

I got the nomkl tip from this Continuum blog [post](https://www.continuum.io/blog/developer-blog/anaconda-25-release-now-mkl-optimizations).

amueller commented:
I think a better fix would be to skip the test, but either way is fine with me. It's most important to fix master.

The test depends on 1e-17 not being zero after multiple floating operations. That doesn't seem to be a good idea.

lesteve commented:
> I think a better fix would be to skip the test, but either way is fine with me. It's most important to fix master.

Note : on top of getting master green, this PR is a fix for properly taking into account INSTALL_MKL=false. Since anaconda 2.5 (early February) not listing mkl in the list of packages to install does not warranty that you are not going to get mkl installed, for example:

```
❯ conda create -n del python=3.5 numpy
Fetching package metadata: ....
Solving package specifications: .............
Package plan for installation in environment /home/lesteve/miniconda3/envs/del:

The following NEW packages will be INSTALLED:

    mkl:        11.3.1-0     
    numpy:      1.10.4-py35_1
    openssl:    1.0.2g-0     
    pip:        8.1.0-py35_0 
    python:     3.5.1-0      
    readline:   6.2-2        
    setuptools: 20.2.2-py35_0
    sqlite:     3.9.2-0      
    tk:         8.5.18-0     
    wheel:      0.29.0-py35_0
    xz:         5.0.5-1      
    zlib:       1.2.8-0 
```

lesteve commented:
Grrr now the scipy-dev-wheels is failing for some reason ...

amueller commented:
unrelated, though.

lesteve commented:
> unrelated, though.

Completely unrelated. Still that means master is not going to get green just by merging this PR ...

lesteve committed:
MAINT temporary disable MKL in Travis

to have master green on Travis until we fix #6279.

Install nomkl when INSTALL_MKL is not true.
DISTRIB=scipy-dev-wheels uses allow_failures.

lesteve commented:
I put the scipy-dev-wheels in allow_failures and at least now Travis is green.

lesteve commented:
FYI I opened a scipy issue related to the scipy-dev-wheels failure https://github.com/scipy/scipy/issues/5948.

ogrisel commented:
+1 for merge as a temporary stop gap.

> The test depends on 1e-17 not being zero after multiple floating operations. That doesn't seem to be a good idea.

I agree.

MechCoder commented:
I'll merge?

Are we solving both the stability issue and the new way in which the MKL package is shipped with the same stone?

MechCoder commented:
It seems that some recent update in scipy causes the accuracy score to return a singleton object..

lesteve commented:
> Are we solving both the stability issue and the new way in which the MKL package is shipped with the same stone? 

* This PR mostly makes sure that INSTALL_MKL is properly taken into account.
* Not installing mkl is a temporary work-around for the PLS test to pass until we figure out what to do about it.
* Putting scipy-dev-wheels in allow_failures is a way to get Travis green even if the scipy-dev-wheels fails. There was a scipy issue (.mean was returning a matrix instead of scalar for sparse matrices) that I fixed. Unfortunately now it is the turn of numpy to misbehave.

MechCoder commented on .travis.yml:
```diff
@@ -39,11 +39,16 @@ env:
+  allow_failures:
+    # allow_failures seems to be keyed on the python version
+    # We are using this to allow failures for DISTRIB=scipy-dev-wheels
+    - python: 3.5
```
We can remove this now because the SciPy problem is fixed?

lesteve replied:
Unfortunately not as I said earlier there is still the numpy problem where the score is a 0d memmap instead of a scalar.

MechCoder commented:
Yes, it seems to be how memmap operations are handled in NumPy. Should I open an issue for that?

MechCoder commented:
Oh, just saw https://github.com/numpy/numpy/issues/7403 . Sorry for cross-posting.

lesteve commented:
Note to my future self and possibly others: when we reenable mkl for the Python 3.5 distrib we want to use numpy=1.10.4. It looks like something is broken with conda and mkl for numpy=1.10.2.

This is what causing the errors like this:
```
Intel MKL FATAL ERROR: Cannot load libmkl_avx.so or libmkl_def.so.
```

ogrisel commented:
Please feel free to upgrade directly to numpy 1.10.4 and scipy 0.17.0 in this PR. There is not point in running the tests on not so recent numpy / scipy for the entry named "recent".

ogrisel commented:
Ok let's merge for now.
